ja/core-libraries/collections.rst

```diff
@@ -1,23 +1,1085 @@
+
+ポジションはゼロが基準なので、最初のポジション番号は ``0`` です。
+
+.. php:method:: skip(int $positions)
+
+``take()`` の第二引数は、コレクションから取得する前にいくつかの要素をスキップすることができますが、
+特定のポジションの後にある残りの要素を取る方法として、同じ目的のために ``skip()`` を使用できます。 ::
+
+    $collection = new Collection([1, 2, 3, 4]);
+    $allExceptFirstTwo = $collection->skip(2)->toList(); // [3, 4]
+
+.. php:method:: first()
+
+``take()`` の最も一般的な用途の1つは、コレクションの最初の要素を取得することです。
+同じ目標を達成するためのショートカットメソッドとして ``first()`` メソッドを使用しています。 ::
+
+    $collection = new Collection([5, 4, 3, 2]);
+    $collection->first(); // 戻り値は 5
+
+.. php:method:: last()
+
+同様に、``last()`` メソッドを使用して、コレクションの最後の要素を取得することができます。 ::
+
+    $collection = new Collection([5, 4, 3, 2]);
+    $collection->last(); // 戻り値は 2
+
+コレクションの拡張
+------------------
+
+.. php:method:: append(array|Traversable $items)
+
+複数のコレクションから１つのコレクションを作成することができます。
+これは、さまざまなソースからデータを収集し、それを連結し、
+非常にスムーズに他のコレクション関数を適用することができます。
+``append()`` メソッドは両方のソースの値を含む新しいコレクションを返します。 ::
+
+    $cakephpTweets = new Collection($tweets);
+    $myTimeline = $cakephpTweets->append($phpTweets);
+
+    // 両方のソースから cakefest を含むつぶやき
+    $myTimeline->filter(function ($tweet) {
+        return strpos($tweet, 'cakefest');
+    });
+
+.. warning::
+
+    異なるソースから追加するときは、両方のコレクションのいくつかのキーが同じこともありえます。
+    例えば、2つの単純な配列を付加します。これは、 ``toArray()`` を使用してコレクションを
+    配列に変換するときに問題を示すことができます。あるコレクションの値で、キーを基にして
+    以前のコレクションの値を上書きしたくないなら、キーを削除して、すべての値を保持するために
+    ``toList()`` を呼び出すことを確認してください。
+
+要素の更新
+----------
+
+.. php:method:: insert(string $path, array|Traversable $items)
+
+時には、２つの別々のデータの集合があり、一方の集合の要素を、
+他方のそれぞれの要素に挿入したいこともあるでしょう。もともとデータのマージや結合を
+サポートしないデータソースからデータを取得する際に非常に一般的なケースです。
+
+あるコレクションの各要素を別のコレクションの各要素のプロパティに挿入することができる
+``insert()`` メソッドを提供します。 ::
+
+    $users = [
+        ['username' => 'mark'],
+        ['username' => 'juan'],
+        ['username' => 'jose']
+    ];
+
+    $languages = [
+        ['PHP', 'Python', 'Ruby'],
+        ['Bash', 'PHP', 'Javascript'],
+        ['Javascript', 'Prolog']
+    ];
+
+    $merged = (new Collection($users))->insert('skills', $languages);
+
+配列に変換すると、 ``$merged`` コレクションは、次のようになります。 ::
+
+    [
+        ['username' => 'mark', 'skills' => ['PHP', 'Python', 'Ruby']],
+        ['username' => 'juan', 'skills' => ['Bash', 'PHP', 'Javascript']],
+        ['username' => 'jose', 'skills' => ['Javascript', 'Prolog']]
+    ];
+
+``insert()`` メソッドの最初のパラメータは、要素がその位置に挿入することができるように示した
+プロパティのドット区切りのパスです。第２引数は、コレクションオブジェクトに変換することができるものです。
+
+要素が順番に挿入されていることを確認してください。第２のコレクションの最初の要素は、
+第１のコレクションの最初の要素にマージされます。
+
+第１のコレクションに挿入する第２のコレクションに十分な要素が存在しない場合、
+対象のプロパティは、 ``null`` 値が入力されます。 ::
+
+    $languages = [
+        ['PHP', 'Python', 'Ruby'],
+        ['Bash', 'PHP', 'Javascript']
+    ];
+
+    $merged = (new Collection($users))->insert('skills', $languages);
+
+    // 結果
+    [
+        ['username' => 'mark', 'skills' => ['PHP', 'Python', 'Ruby']],
+        ['username' => 'juan', 'skills' => ['Bash', 'PHP', 'Javascript']],
+        ['username' => 'jose', 'skills' => null]
```

Comment:
This example seems incorrect to me. In my environment, the key `skills` didn't appear. It doesn't relate to translating though.